src/core/view/StrokeView.cpp

```diff
@@ -118,19 +118,25 @@ void StrokeView::draw(const Context& ctx) const {
         // We assume the matrix is diagonal (i.e. only scaling, no rotation)
         assert(matrix.xy == 0 && matrix.yx == 0);
 
+        double scaleX;
+        double scaleY;
+        cairo_surface_get_device_scale(cairo_get_target(ctx.cr), &scaleX, &scaleY);
```

Comment:
I would add a comment along the lines of "The mask needs to be scaled by both zoom and device scale (e.g., to be same as the canvas), otherwise the stroke will rendered at a lower resolution". This comment could be combined with the one a few ones above.

I also think the code here would be easier to understand if you moved this snippet above the matrix snippet, so that it's clear that `scaleX, scaleY` refer to the final scale values:
```c++
Define the scale variables
Initialize them to the device scale
Get the transformation matrix
Multiply the scale values by the matrix values
```